plugin/src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBaseExtension.kt

```diff
@@ -69,11 +71,9 @@ abstract class MavenPublishBaseExtension(
       repo.credentials(PasswordCredentials::class.java)
     }
 
-    project.gradlePublishing.publications.withType(MavenPublication::class.java).all { publication ->
-      project.afterEvaluate {
-        project.tasks.named("publish${publication.name.capitalize()}PublicationToMavenCentralRepository") {
-          it.dependsOn(createRepository)
-        }
+    project.tasks.withType(PublishToMavenRepository::class.java).configureEach { publishTask ->
+      if (publishTask.name.endsWith("ToMavenCentralRepository")) {
+        publishTask.dependsOn(createRepository)
```

Comment:
Found out that `PublishToMavenRepository` is in the public API so we can use it here as a filter which makes it nicer than afterEvaluate